src/middlewares/__tests__/createInsightsMiddleware.ts

```diff
@@ -114,6 +117,19 @@ describe('insights', () => {
       });
     });
 
+    it('throws when search client does not have credentials', () => {
+      const { insightsClient, instantSearchInstance } = createTestEnvironment({
+        searchClient: createSearchClient(),
+      });
+      expect(() =>
+        createInsightsMiddleware({
+          insightsClient,
+        })({ instantSearchInstance })
+      ).toThrowErrorMatchingInlineSnapshot(
+        `"[insights middleware]: could not extract Algolia credentials from searchClient"`
```

Comment:
Seems reasonable, but I'm curious in which case searchClient didn't include the credentials? In that case, InstantSearch.js itself won't be functioning, no?

Reply:
if you do the following:

```
const client = algoliasearch()
const searchClient = {
  search: client.search // or intercept
}
```

In that case only the "required" methods are present, as well as with a "backend search" without an algolia client the insights-thrown error is confusing

Reply:
Ah okay, that makes sense.